server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -1023,6 +1024,31 @@ protected Node(
             // Add the telemetryAwarePlugin components to the existing pluginComponents collection.
             pluginComponents.addAll(telemetryAwarePluginComponents);
 
+            Collection<Object> indicesAwarePluginComponents = pluginsService.filterPlugins(IndicesAwarePlugin.class)
+                .stream()
+                .flatMap(
+                    p -> p.createComponents(
+                        client,
+                        clusterService,
+                        threadPool,
+                        resourceWatcherService,
+                        scriptService,
+                        xContentRegistry,
+                        environment,
+                        nodeEnvironment,
+                        namedWriteableRegistry,
+                        clusterModule.getIndexNameExpressionResolver(),
+                        repositoriesServiceReference::get,
+                        tracer,
+                        metricsRegistry,
```

Comment:
Why are we adding `tracer` and `metricsRegistry`?

Reply:
`metricsRegistry` is needed in `QueryInsightsPlugin`. `tracer` is uneccessary

Reply:
But, it does not need to be part of the same plugin. `QueryInsightsPlugin` can implement `createComponents` method from both `TelemetryAwarePlugin` and `MappingAwarePlugin`. The coupling between `metricsRegistry` and `mapperServiceProvider` just because of `QueryInsightsPlugin` does not make sense to me!

Reply:
Got it now, createComponents() will get called twice for QueryInsightsPlugin with different set of args each time.